cmd/pfcpsim-client/main.go

```diff
@@ -3,6 +3,362 @@
+		}
+		remotePeerAddress = net.ParseIP(address[0])
+	}
+
+	upfAddress = net.ParseIP(*upfAddr)
+	if upfAddress == nil {
+		log.Fatalf("Error while parsing UPF address")
+	}
+
+	var err error = nil
+
+	_, _, err = net.ParseCIDR(*ueAddrPool)
+	if err != nil {
+		log.Fatalf("Could not parse ue address pool: %v", err)
+	}
+	ueAddressPool = *ueAddrPool
+}
+
+// readInput will cycle through user's input. if inputFile was provided as a flag, Stdin redirection is performed.
+func readInput(input chan<- string) {
+	if inputFile != "" {
+		// Set inputFile as stdIn
+
+		oldStdin := os.Stdin
+		defer func() {
+			// restore StdIN
+			os.Stdin = oldStdin
+		}()
+
+		f, err := os.Open(inputFile)
+		if err != nil {
+			log.Errorf("Error while reading inputFile: %v", err)
+		} else {
+			defer func(f *os.File) {
+				err := f.Close()
+				if err != nil {
+					log.Errorf("Error while closing file: %v", err)
+				}
+			}(f)
+
+			os.Stdin = f
+		}
+	}
+
+	for {
+		var u string
+		_, err := fmt.Scanf("%s\n", &u)
+		if err != nil {
+			if err == io.EOF {
+				break
+			} else {
+				log.Debugf("Skipping bad entry: %v", err)
+			}
+		}
+		input <- u
+	}
+}
+
+// handleUserInput spawn a goroutine cycling through user's input.
+func handleUserInput() {
+	userInput := make(chan string)
+	go readInput(userInput)
+
+	for {
+		fmt.Println("'disassociate': Teardown Association")
+		fmt.Println("'associate': Setup Association")
+		fmt.Println("'create': Create Sessions ")
+		fmt.Println("'delete': Delete Sessions ")
+		fmt.Println("'exit': Exit ")
+		fmt.Print("Enter service: ")
+
+		select {
+		case userAnswer := <-userInput:
+			switch userAnswer {
+			case "disassociate":
+				log.Tracef("Selected teardown association")
+				err := globalPFCPSimClient.TeardownAssociation()
+				if err != nil {
+					log.Errorf("Error while tearing down association: %v", err)
+				}
+
+			case "associate":
+				log.Tracef("Selected setup association")
+				err := globalPFCPSimClient.SetupAssociation()
+				if err != nil {
+					log.Errorf("Error while setting up association: %v", err)
+				}
+
+			case "create":
+				log.Tracef("Selected create sessions")
+				InitializeSessions(sessionCount)
+
+			case "delete":
+				log.Tracef("Selected delete sessions")
+				err := globalPFCPSimClient.DeleteAllSessions()
+				if err != nil {
+					log.Errorf("Error while deleting sessions: %v", err)
+				}
+
+			case "exit":
+				log.Tracef("Shutting down")
+				globalPFCPSimClient.DisconnectN4()
+				os.Exit(0)
+
+			default:
+				log.Error("Command not found")
+			}
+		}
+	}
+}
+
+// getNextUEAddress retrieves the next available IP address from ueAddressPool
+func getNextUEAddress() net.IP {
+	// TODO handle case net IP is full
+	if lastUEAddress == nil {
+		ueIpFromPool, _, _ := net.ParseCIDR(ueAddressPool)
+		lastUEAddress = iplib.NextIP(ueIpFromPool)
+
+		return lastUEAddress
+
+	} else {
+		lastUEAddress = iplib.NextIP(lastUEAddress)
+		return lastUEAddress
+	}
```

Comment:
Avoid using if -> else:

```suggestion
	if lastUEAddress != nil {
	        lastUEAddress = iplib.NextIP(lastUEAddress)
		return lastUEAddress
	}
	
	// TODO handle case net IP is full
	ueIpFromPool, _, _ := net.ParseCIDR(ueAddressPool)
	lastUEAddress = iplib.NextIP(ueIpFromPool)

	return lastUEAddress
```